File: backend/schemas/admin_division.py
# backend/schemas/admin_division.py
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel


class CountryOut(BaseModel):
    code: str
    name: str
    source: str
    adminLevels: int


class CountryListOut(BaseModel):
    items: List[CountryOut]


class AdminDivisionOut(BaseModel):
    id: UUID
    countryCode: str
    level: int
    code: str
    name: str
    parentId: Optional[UUID] = None
    source: str
    # URI estable para dwc:locationID (ubigeo o GeoNames)
    locationId: str


class AdminDivisionListOut(BaseModel):
    items: List[AdminDivisionOut]

File: frontend/src/interfaces/adminDivision.ts
// Divisiones administrativas (catálogo INEI + GeoNames)
export interface CatalogCountry {
  code: string;
  name: string;
  source: "INEI" | "GEONAMES";
  adminLevels: number;
}

export interface AdminDivision {
  id: string;
  countryCode: string;
  level: number; // 1 departamento · 2 provincia · 3 distrito
  code: string;
  name: string;
  parentId: string | null;
  source: "INEI" | "GEONAMES";
  /** URI estable para dwc:locationID (ubigeo o GeoNames) */
  locationId: string;
}
